### PR TITLE
feature(proxy): allow multiple routes per endpoint

### PR DESCRIPTION
**Pull request by ondrej-til**
- multiple/alternative routes for a single endpoint is now supported
- set alternative routes for o365 calendar

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -69,23 +69,31 @@ const bootstrap = async () => {
...
-        const path = `/${host}${route.path}`
-        const method = route.method || 'get'
-        const requireAuth = route.requireAuth !== false
-
-        console.info(`Registering route [${method.toUpperCase()}] ${path}`)
-
-        const handlers = []
-        if (Array.isArray(route.handler)) {
-          handlers.push(...route.handler)
+        const paths = []
```
Comment (janjilek) [inline src/main.ts:72]: I would change it to:
```typescript
const paths = Array.isArray(route.path)
  ? route.path
  : [route.path]

paths
   .forEach(routePath => {
    const path = `/${host}${routePath}`
    // content of the `for (let path of paths) {}` loop
   })
```

File: src/main.ts [hunk 2]
```diff
@@ -69,23 +69,31 @@ const bootstrap = async () => {
...
+        if (route.path instanceof Array) {
```
Comment (janjilek) [inline src/main.ts:73]: Better to use `Array.isArray`

File: src/main.ts [hunk 3]
```diff
@@ -69,23 +69,31 @@ const bootstrap = async () => {
...
+          paths.push(`/${host}${route.path}`)
+        }
+        console.log("PATHS " + paths)
+        for (let path of paths) {
+          const method = route.method || 'get'
+          const requireAuth = route.requireAuth !== false
+
+          console.info(`Registering route [${method.toUpperCase()}] ${path}`)
+
+          const handlers = []
```
Comment (janjilek) [inline src/main.ts:85]: We should change it to the same format as `paths` for the sake of consistency:
```typescript
const handlers = Array.isArray(route.handler)
  ? route.handler
  : [route.handler]
```

File: src/router/interfaces/router.interface.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { RequestHandler } from 'express'
 
 export interface Route {
   hosts: string[],
-  path: string,
+  path: string|string[],
```
Comment (janjilek) [inline src/router/interfaces/router.interface.ts:5]: use space between types `string | string[]`

File: src/main.ts [hunk 5]
```diff
@@ -69,23 +69,31 @@ const bootstrap = async () => {
...
-        if (Array.isArray(route.handler)) {
-          handlers.push(...route.handler)
+        const paths = []
+        if (route.path instanceof Array) {
+          route.path.forEach((routePath) => paths.push(`/${host}${routePath}`))
         } else {
-          handlers.push(route.handler)
+          paths.push(`/${host}${route.path}`)
+        }
+        console.log("PATHS " + paths)
```
Comment (janjilek) [inline src/main.ts:78]: This should be removed

Comment (ondrej-til) [reply src/main.ts:78]: done

**Pull request merged**